src/gt4py/next/otf/compilation/build_systems/cmake_lists.py

```diff
@@ -93,13 +99,24 @@ def visit_LinkDependency(self, dep: LinkDependency):
         match dep.name:
             case "pybind11":
                 lib_name = "pybind11::module"
+            case "nanobind":
+                lib_name = "nanobind-static"
             case "gridtools":
                 lib_name = "GridTools::fn_naive"
             case _:
                 raise ValueError("Library {name} is not supported".format(name=dep.name))
-        return "target_link_libraries({target} PUBLIC {lib})".format(
-            target=dep.target, lib=lib_name
-        )
+
+        cfg = ""
+        if dep.name == "nanobind":
+            cfg = "\n".join(
+                [
+                    "nanobind_build_library(nanobind-static)",
+                    f"nanobind_compile_options({dep.target})",
+                    f"nanobind_link_options({dep.target})",
+                ]
+            )
```

Comment:
is this because nanobind's doesn't expose a proper cmake target?

Reply:
I haven't looked into it's source code to see how it works exactly, but this is what the documentation suggests.